go_build_process.go

```diff
@@ -94,22 +93,22 @@ func (p GoBuildProcess) Execute(config GoBuildConfiguration) (string, error) {
 		p.logs.Action("Failed after %s", duration.Round(time.Millisecond))
 		p.logs.Detail(buffer.String())
 
-		return "", fmt.Errorf("failed to execute 'go build': %w", err)
+		return nil, fmt.Errorf("failed to execute 'go build': %w", err)
 	}
 
 	p.logs.Action("Completed in %s", duration.Round(time.Millisecond))
 	p.logs.Break()
 
 	paths, err := filepath.Glob(fmt.Sprintf("%s/*", config.Output))
```

Comment:
Now that the binaries are exposed as launch processes I think we need to provide the ability for buildpack consumers to control which one is the default process. I think the default process needs to be the first target listed in the `BP_GO_TARGETS` rather than just the first binary in the glob search. This decision is likely a breaking change since the buildpack today will select the first alphabetical globbed binary.

Reply:
If I have
```
```go-multiple-targets/
├── cmd
│   ├── one
│   │   └── main.go
│   └── two
│       └── main.go
├── go.mod
├── main.go
```

Then

`BP_GO_TARGETS="./cmd/two:.:./cmd/one"`
`BP_GO_TARGETS=".:./cmd/one:./cmd/two"`
`BP_GO_TARGETS="./cmd/one:./cmd/two:."`

should all provide a different default launch process

Reply:
Hmm I see what you mean. I will try and fix this up.

Reply:
@wburningham Alright after some fettling, I have managed to get all of the paths in order of appearance meaning the first target's binary should be the default process type. Take a look and see if that works for you?